WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCoupon.swift

```diff
@@ -15,29 +15,196 @@ struct AddEditCoupon: View {
 
     var body: some View {
         NavigationView {
+            GeometryReader { geometry in
+                ScrollView {
+                    VStack (alignment: .leading) {
+                        Group {
+                            ListHeaderView(text: Localization.headerCouponDetails.uppercased(), alignment: .left)
 
-            //TODO: implement the content of the view
-            Text("Hello, World!")
-                .toolbar {
-                                    ToolbarItem(placement: .cancellationAction) {
-                                        Button("Cancel", action: {
-                                            presentation.wrappedValue.dismiss()
-                                        })
-                                    }
-                                }
+                            Group {
+                                TitleAndTextFieldRow(title: Localization.couponAmountPercentage,
+                                                     placeholder: Localization.couponAmountPercentage,
+                                                     text: $viewModel.amountField,
+                                                     editable: false,
+                                                     fieldAlignment: .leading,
+                                                     keyboardType: .asciiCapableNumberPad)
```

Comment:
I imagine this PR only focuses on the UI, so maybe the texts are not finalized. Just want to point out that the title for the Amount may need to consider the coupon discount type, so that we can display either "Amount (%)" and "Amount (currency symbol)" depending on the type.

Regarding the placeholder, I think we should display either a potential value (e.g "0" in this case), or a message (e.g "enter amount"). Duplicating the title makes it look a bit redundant - WDYT?

Also - the field should be editable right?

And the keyboard type should be `.decimalPad` since we may want to be able to enter floating number as well.

Reply:
> I imagine this PR only focuses on the UI, so maybe the texts are not finalized. Just want to point out that the title for the Amount may need to consider the coupon discount type, so that we can display either "Amount (%)" and "Amount (currency symbol)" depending on the type.

Absolutely yes, It's something I'm handling in a separate PR 👍 

> Regarding the placeholder, I think we should display either a potential value (e.g "0" in this case), or a message (e.g "enter amount"). Duplicating the title makes it look a bit redundant - WDYT?

Same, I'm handling the placeholder in a separate PR

> Duplicating the title makes it look a bit redundant

I agree, it was just a placeholder of placeholder 🤣 

> Also - the field should be editable right?
> And the keyboard type should be .decimalPad since we may want to be able to enter floating number as well.

Yes I'm handling all these small details in a separate PR, sorry for not saying this in the PR description 🙇